File: src/main/java/com/tibiawiki/domain/repositories/ArticleRepository.java
package com.tibiawiki.domain.repositories;

import org.fastily.jwiki.core.NS;

import java.util.List;
import java.util.Map;

public interface ArticleRepository {

    List<String> getPageNamesFromCategory(String categoryName);

    List<String> getPageNamesFromCategory(String categoryName, NS namespace);

    Map<String, String> getArticlesFromCategory(List<String> pageNames);

    Map<String, String> getArticlesFromCategory(String categoryName);

    List<String> getPageNamesUsingTemplate(String templateName);

    String getArticle(String pageName);

    boolean modifyArticle(String pageName, String pageContent, String editSummary);
}

File: src/main/kotlin/com/tibiawiki/domain/ArticleRepository.kt
package com.tibiawiki.domain

import org.fastily.jwiki.core.NS

interface ArticleRepository {

    fun getPageNamesFromCategory(categoryName: String): List<String>

    fun getPageNamesFromCategory(categoryName: String, namespace: NS): List<String>

    fun getArticlesFromCategory(pageNames: List<String>): Map<String, String>

    fun getArticlesFromCategory(categoryName: String): Map<String, String>

    fun getPageNamesUsingTemplate(templateName: String): List<String>

    fun getArticle(pageName: String): String?

    fun modifyArticle(pageName: String, pageContent: String, editSummary: String): Boolean
}
